src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -42,8 +49,51 @@ private static void initHttpClient()
         null,
         new DefaultHostnameVerifier());
 
+
+    /**
+     * Use a anonymous class to implement
+     * the interface ServiceUnavailableRetryStrategy()
+     * The max retry time is 3.
+     * The interval time is backoff.
+     */
     httpClient = HttpClients.custom()
         .setSSLSocketFactory(f)
+        .setServiceUnavailableRetryStrategy(new ServiceUnavailableRetryStrategy()
+        {
+          private int executionCount = 0;
+          @Override
+          public boolean retryRequest(
+              final HttpResponse response, final int executionCount,
+              final HttpContext context)
+          {
+            this.executionCount = executionCount;
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            boolean needNextRetry = (statusCode == 408 || statusCode >= 500 )
+                && executionCount < 4;
+            if(executionCount == 4)
+            {
+              LOGGER.info("Reach the max retry time.");
+            }
+
+            if(needNextRetry && executionCount < 4)
+            {
+              long interval = (1 << executionCount)  * 1000;
+              LOGGER.info("Will retry in " +
+                  String.valueOf(interval) + " millisecond.");
+            }
+
+            return needNextRetry;
+          }
+
+          @Override
+          public long getRetryInterval()
+          {
+            long interval = (1 << executionCount)  * 1000;
```

Comment:
Add a comment above here indicating what exactly you are doing for the retry interval. Also add a small comment indicating that the retryInterval is in milli seconds.